File: app/src/main/java/tk/lorddarthart/rxnewstestapp/JSONPlaceHolderApi.java
package tk.lorddarthart.rxnewstestapp;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface JSONPlaceHolderApi {
    @GET("/")
    Call<News> getPostWithID();
}

File: app/src/main/java/tk/lorddarthart/rxnewstestapp/util/JSONPlaceHolderApi.kt
package tk.lorddarthart.rxnewstestapp.util

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query
import tk.lorddarthart.rxnewstestapp.application.model.News

interface JSONPlaceHolderApi {
    @GET("/uc")
    fun getPostWithID(@Query(value = "id", encoded = true) id: String): Call<News>
}
